rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp

```diff
@@ -187,6 +187,12 @@ bool TimeControllerClock::sleep_until(rclcpp::Time until)
   return sleep_until(until.nanoseconds());
 }
 
+void TimeControllerClock::wakeup()
+{
+  std::lock_guard<std::mutex> lock(impl_->state_mutex);
+  impl_->cv.notify_all();
```

Comment:
Not 100% sure, but it seems we don't need to lock a mutex here.

Reply:
That does seem to be right: https://en.cppreference.com/w/cpp/thread/condition_variable/notify_all

> The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s). Doing so may be a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock, though some implementations recognize the pattern and do not attempt to wake up the thread that is notified under lock.

Removed: 9464665401d9ab05e7cd9980accafafa98b02803